src/App.tsx

```diff
@@ -1,61 +1,116 @@
+import { ChangeEvent, FormEvent, useState } from 'react';
+import { TodoList } from './components/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const updatedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId) || null,
+}));
+
+export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>(updatedTodos);
+  const [title, setTitle] = useState('');
+  const [userName, setUserName] = useState(0);
+
+  const [isTitleError, setTitleError] = useState(false);
+  const [isUserError, setUserError] = useState(false);
+
+  const newTodoId = Math.max(...todos.map(todo => todo.id));
```

Comment:
```suggestion
  const newTodoId = Math.max(...todos.map(todo => todo.id)) + 1;
```
Better save the final version of new id